client/src/test/java/io/pravega/client/admin/impl/StreamManagerImplTest.java

```diff
@@ -865,4 +870,93 @@ public void testListCompletedTransactions() throws ConnectionFailedException {
+            Long distance = cf.join();
+            assertEquals(Long.valueOf(40), distance);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testGetDistanceBetweenTwoSCWithEndSCUnbounded() throws ConnectionFailedException {
+        final String streamName = "stream";
+        final Stream stream = new StreamImpl(defaultScope, streamName);
+
+        // Setup Mocks
+        ClientConnection connection = mock(ClientConnection.class);
+        PravegaNodeUri location = new PravegaNodeUri("localhost", 0);
+        Mockito.doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                WireCommands.CreateSegment request = (WireCommands.CreateSegment) invocation.getArgument(0);
+                connectionFactory.getProcessor(location)
+                        .process(new WireCommands.SegmentCreated(request.getRequestId(), request.getSegment()));
+                return null;
+            }
+        }).when(connection).send(Mockito.any(WireCommands.CreateSegment.class));
+
+        Mockito.doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                WireCommands.GetStreamSegmentInfo request = (WireCommands.GetStreamSegmentInfo) invocation.getArgument(0);
+                connectionFactory.getProcessor(location)
+                        .process(new WireCommands.StreamSegmentInfo(request.getRequestId(), request.getSegmentName(), true,
+                                false, false, 0, 20, 0));
+                return null;
+            }
+        }).when(connection).send(Mockito.any(WireCommands.GetStreamSegmentInfo.class));
+        connectionFactory.provideConnection(location, connection);
+        MockController mockController = new MockController(location.getEndpoint(), location.getPort(),
+                connectionFactory, true);
+        ConnectionPoolImpl pool = new ConnectionPoolImpl(ClientConfig.builder().maxConnectionsPerSegmentStore(1).build(), connectionFactory);
+        @Cleanup
+        final StreamManager streamManager = new StreamManagerImpl(mockController, pool);
+
+        streamManager.createScope(defaultScope);
+        streamManager.createStream(defaultScope, streamName, StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build());
+        final StreamCut startStreamCut = getStreamCut(defaultScope, streamName, 10L, 0, 1);
+        final StreamCut endStreamCut = StreamCut.UNBOUNDED;
+
+        CompletableFuture<Long> cf = streamManager.getDistanceBetweenTwoStreamCuts(stream, startStreamCut,
+                endStreamCut);
+        try {
+            Long distance = cf.join();
+            assertEquals(Long.valueOf(20), distance);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
```

Comment:
Please also add test coverage for the other two cases:
* Start cut at UNBOUNDED and endCut at 10 when stream ends at 30.
* Start cut at UNBOUNDED and endcut at 30 but stream is truncated to 10.

Reply:
Added two new tests as suggested.